protoc-gen-gogo/generator/generator.go

```diff
@@ -1428,6 +1428,10 @@ func (g *Generator) generateEnum(enum *EnumDescriptor) {
 		g.PrintComments(fmt.Sprintf("%s,%d,%d", enum.path, enumValuePath, i))
 
 		name := ccPrefix + *e.Name
+		if gogoproto.IsEnumCustomName(e) {
```

Comment:
I'll try to match the existing precedence choices if possible. As a user, I would think the more specific customname (enum value) would always override a less-specific (enum) prefix. I'll make sure there is a test case for this.